src/puppetlabs/puppetdb/command.clj

```diff
@@ -456,28 +458,30 @@
 (defn exec-command
   "Takes a command object and processes it to completion. Dispatch is
    based on the command's name and version information"
-  [{:keys [command version] :as cmd} db conn-status start]
+  ([{:keys [command version] :as cmd} db conn-status start]
+    (exec-command cmd db conn-status start {}))
+  ([{:keys [command version] :as cmd} db conn-status start options-config]
   (when-not (supported-command? cmd)
     (throw (ex-info (trs "Unsupported command {0} version {1}" command version)
                     {:kind ::fatal-processing-error})))
   (let [exec (case command
                "replace catalog" exec-replace-catalog
                "replace facts" exec-replace-facts
-               "store report" exec-store-report
+               "store report" (partial exec-store-report options-config)
                "deactivate node" exec-deactivate-node
                "configure expiration" exec-configure-expiration
                "replace catalog inputs" exec-replace-catalog-inputs)]
-    (exec cmd start db conn-status)))
+    (exec cmd start db conn-status))))
 
 (defn process-command!
   ;; only used by testing...
   "Takes a command object and processes it to completion. Dispatch is
    based on the command's name and version information"
-  [command db blocklist-config]
+  [command db options-config]
   (when-not (:delete? command)
     (let [start (now)]
       (-> command
-          (prep-command blocklist-config)
+          (prep-command options-config)
           (exec-command db (atom {}) start)))))
```

Comment:
For `store-report` we need to pass the options-config to `exec-command` as well.